src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Data/ReadyToRunInfo.cs

```diff
@@ -18,7 +18,14 @@ public ReadyToRunInfo(Target target, TargetPointer address)
         CompositeInfo = target.ReadPointer(address + (ulong)type.Fields[nameof(CompositeInfo)].Offset);
 
         NumRuntimeFunctions = target.Read<uint>(address + (ulong)type.Fields[nameof(NumRuntimeFunctions)].Offset);
-        RuntimeFunctions = target.ReadPointer(address + (ulong)type.Fields[nameof(RuntimeFunctions)].Offset);
+        RuntimeFunctions = NumRuntimeFunctions > 0
+            ? target.ReadPointer(address + (ulong)type.Fields[nameof(RuntimeFunctions)].Offset)
+            : TargetPointer.Null;
+
+        NumHotColdMap = target.Read<uint>(address + (ulong)type.Fields[nameof(NumHotColdMap)].Offset);
+        HotColdMap = NumHotColdMap > 0
```

Comment:
Based on the previous description of the HotColdMap, can we assert that the count is even?